src/alvarium/annotators/pki_http.py

```diff
@@ -0,0 +1,87 @@
+import socket
+import os
+from alvarium.annotators.handler.contracts import HTTP_REQUEST_KEY
+from alvarium.annotators.handler.exceptions import ParserException
+
+from alvarium.sign.exceptions import SignException
+from alvarium.sign.factories import SignProviderFactory
+from alvarium.contracts.annotation import Annotation, AnnotationType
+from alvarium.hash.contracts import HashType
+from alvarium.sign.contracts import KeyInfo, SignInfo, KeyInfo, SignType
+from alvarium.utils import PropertyBag
+from .contracts import Signable
+from .utils import derive_hash, sign_annotation
+from .interfaces import Annotator
+from .exceptions import AnnotatorException
+from alvarium.annotators.handler.utils import parseSignature
+
+class HttpPkiAnnotator(Annotator):
+    def __init__(self, hash: HashType, sign_info: SignInfo) -> None:
+        self.hash = hash
+        self.sign_info = sign_info
+        self.kind = AnnotationType.PKIHTTP
+
+    def execute(self, data: bytes, ctx: PropertyBag = None) -> Annotation:
+        key = derive_hash(hash=self.hash, data=data)
+        host: str = socket.gethostname()
+
+        # Call parser on request
+        req = ctx.get_property(HTTP_REQUEST_KEY)
+
+        try:
+            parsed_data = parseSignature(req)
+        except ParserException as e:
+            raise AnnotatorException("Cannot parse the HTTP request.", e)
+
+        signable = Signable(parsed_data.seed, parsed_data.signature)
```

Comment:
Just an FYI, It's always better to use named arguments instead of positional arguments in these cases. Suppose that inside the `Signable` class, we've interchanged the order of arguments to be `signature` then `seed`. It will **not** throw an error as both entries are valid strings. This bug will be really hard to debug, so I suggest that you always favor named arguments over positional ones. (that applies to method call or object instantiation across the SDK)